pyglmnet/pyglmnet.py

```diff
@@ -175,9 +175,22 @@ def _grad_L2loss(distr, alpha, Tau, reg_lambda, X, y, eta, beta):
                    ((1 - y) * (grad_prob / (1 - prob))))
         grad_logl = ((y * (grad_prob / prob)) -
                      ((1 - y) * (grad_prob / (1 - prob))))
-        grad_beta = -1. / n_samples * np.transpose(np.dot(grad_logl.T, X)) + \
-            reg_lambda * (1 - alpha) * np.dot(InvCov, beta[1:])
+        grad_beta = -1. / n_samples * np.transpose(np.dot(grad_logl.T, X))
 
+    elif distr == 'gamma':
+        nu = 1.
+
+        def ilink(z):
+            return np.exp(z)
+
+        def grad_ilink(z):
+            return np.exp(z)
+
+        grad_logl = (y / ilink(z) ** 2 + 1 / ilink(z)) * grad_ilink(z)
+        grad_beta0 = - 1. / n_samples * nu * np.sum(grad_logl)
```

Comment:
this part can probably be factored out of the `if` statements so we simplify things. Later ...